DimensionDev/Maskbook
# feat(layer2): add Aurora Optimistic Network

## Description

### Aurora Layer 2
https://doc.aurora.dev/develop/compat/evm https://github.com/aurora-is-near

### Aurora Dex

- Dodoex ✅ 
- WannaSwap ✅ 
- Trisolaris ✅  https://github.com/dimensiondev/maskbook/issues/5818

### Token List
 -  [x] https://github.com/DimensionDev/Mask-Token-List/pull/46 ✅

### Contracts  Mainnet 
- [x] BalanceChecker 
- [x] MultiCall

### Contracts  Testnet
- [x] BalanceChecker  ✅ 
- [x] MultiCall ✅ 
- [x] Maskbook Contracts  ✅ 


### Sourcify
- [ ] https://github.com/ethereum/sourcify/pull/621

### Aurora Bridge
https://rainbowbridge.app/

### Faucet
https://help.aurora.dev/article/22-where-can-i-get-tokens-for-the-testnet

### Setup Metadata

- [x] Add a logo image
  - `packages/mask/src/plugins/EVM/assets/`
- [x] Add metadata data
  - `packages/web3-shared/evm/assets/chains.json`
- [x] Add an HTTP RPC node
  - `packages/web3-constants/evm/rpc.json`
- [x] Set network descriptor
  - `PLUGIN_NETWORKS` in `packages/mask/src/plugins/EVM/constants.ts`

### Setup External APIs

| Provider      | Configuration Path                               | status |
| ------------- | ------------------------------------------------ |-------|
| DeBank        | `packages/web3-constants/evm/debank.json`        | ✅ |
| CoinGecko     | `packages/web3-constants/evm/coingecko.json`     |✅ |
| CoinMarketCap | `packages/web3-constants/evm/coinmarketcap.json` |✅ 
| Zerion        | `packages/web3-constants/evm/zerion.json`        | ✅ |


| DEX       | Pull Request Link                                  |status |
| --------- | -------------------------------------------------- |-------|
| DODO      | https://github.com/DimensionDev/Maskbook/pull/5323/ |✅ |
| WannaSwap |https://github.com/DimensionDev/Maskbook/pull/5323/ |✅ |
| Trisolaris |https://github.com/DimensionDev/Maskbook/pull/5323/ |✅ |



## Testing Checklist

Before we ship the chain to the user, we need to do basic ability checks.

- [x] Check if the asset list and transaction history on the dashboard page work when you choose the chain as the network.
  - Setup the chain for DeBank API.
- [x] Check if the trending view is working. Try to hover a new chain token in any tweets.
  - Setup the chain for Coingecko API.
  - Setup the chain for CoinMarketCap API.

![image](x)

- [x] Check if the gas estimate dialog is working.
  - Setup the chain for CoinGecko API.

Goto `chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/dashboard.html#/wallets/transfer` and check the estimated USD value is working.

![image](x)

- [x] Trade with the DEX on the chain.

  - Integrate a DEX for the chain.

- [x] Send a transaction and check if the explorer links are working.
  - Setup metadata and RPC for the chain.
